assets/src/dashboard/app/views/editorSettings/index.js

```diff
@@ -15,60 +15,160 @@
+
+  const [mediaError, setMediaError] = useState('');
+  /**
+   * WP settings references publisher logos by ID.
+   * We must retrieve the media for those ids from /media when present
+   * Summarized below:
+   * FETCH - fetch settings to know what IDs to load media for
+   * then fetch media when publisherLogoIds are in state with useEffect
+   * ADD - first upload new logo to media. use the new ID(s) to update settings.
+   * REMOVE - send id to remove to settings. Not deleting media, just removing from logos.
+   */
+
   useEffect(() => {
     fetchSettings();
   }, [fetchSettings]);
 
-  const handleCompleteUpdateId = useCallback(
-    ({ newGoogleAnalyticsId }) => {
-      const updatedSettings = {
-        googleAnalyticsId:
-          typeof newGoogleAnalyticsId === 'string' && newGoogleAnalyticsId,
-      };
+  useEffect(() => {
+    if (newlyCreatedMediaIds.length > 0) {
+      updateSettings({ publisherLogoIds: newlyCreatedMediaIds });
+    }
+  }, [updateSettings, newlyCreatedMediaIds]);
+
+  useEffect(() => {
+    if (publisherLogoIds.length > 0) {
+      fetchMediaById(publisherLogoIds);
+    }
+  }, [fetchMediaById, publisherLogoIds]);
+
+  const handleUpdateGoogleAnalyticsId = useCallback(
+    (newGoogleAnalyticsId) =>
+      updateSettings({ googleAnalyticsId: newGoogleAnalyticsId }),
+    [updateSettings]
+  );
+
+  const handleAddLogos = useCallback(
+    (files) => {
+      const isFileSizeWithinMaxUpload = files.every(
+        (file) => file.size <= maxUpload
+      );
 
-      updateSettings(updatedSettings);
+      if (!isFileSizeWithinMaxUpload) {
+        const errorText =
+          files.length === 1
+            ? sprintf(
+                /* translators: %s: max upload size for media */
+                __(
+                  'Sorry, this file is too big. Make sure your logo is under %s.',
+                  'web-stories'
+                ),
+                maxUploadFormatted
+              )
+            : sprintf(
+                /* translators: %s: max upload size for media */
+                __(
+                  'Sorry, one or more of these files are too big. Make sure your logos are all under %s.',
+                  'web-stories'
+                ),
+                maxUploadFormatted
+              );
+        return setMediaError(errorText);
+      }
+      setMediaError('');
+      return uploadMedia(files);
+    },
+    [maxUpload, maxUploadFormatted, uploadMedia]
+  );
+
+  const handleRemoveLogo = useCallback(
+    (e, media) => {
+      e.preventDefault();
+      updateSettings({ publisherLogoIdToRemove: media.id });
     },
     [updateSettings]
   );
 
+  const orderedPublisherLogos = useMemo(() => {
+    if (Object.keys(mediaById).length <= 0) {
+      return [];
+    }
+    return publisherLogoIds.map((publisherLogoId) => {
+      if (mediaById[publisherLogoId]) {
+        if (publisherLogoId === activePublisherLogoId) {
+          mediaById[publisherLogoId].isActive = true;
```

Comment:
Mutating this will mutate the value back in the provider (and not trigger any updates) and could cause some confusing bugs.  Rather than mutating this here we should create an effect to update the provider state or we can do the following in the the return statement:

```javascript
return publisherLogoId === activePublisherLogoId ? { ...mediaById[publisherLogoId], isActive: true } : mediaById[publisherLogoId];
```

Reply:
great catch, thanks! updating now 🙃